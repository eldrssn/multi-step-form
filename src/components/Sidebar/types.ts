export type TSidebar = {
  step: number;
};
