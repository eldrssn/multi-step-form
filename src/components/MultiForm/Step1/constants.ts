import { TFormStep } from './types';

export const step1: TFormStep[] = [
  {
    name: 'name',
    label: 'Name',
    placeholder: 'e.g. Stephen King',
    type: 'text',
  },
  {
    name: 'email',
    label: 'Email Address',
    placeholder: 'e.g. stephenking@lorem.com',
    type: '',
  },
  {
    name: 'phone',
    label: 'Phone Number',
    placeholder: 'e.g. +1 234 567 890',
    type: 'number',
  },
];
