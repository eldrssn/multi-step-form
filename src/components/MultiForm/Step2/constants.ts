import { TPlan } from './types';

export const plans: TPlan[] = [
  {
    icon: '/src/assets/images/icon-arcade.svg',
    title: 'Arcade',
    priceMonthly: '9',
    priceYearly: '90',
  },
  {
    icon: '/src/assets/images/icon-advanced.svg',
    title: 'Advanced',
    priceMonthly: '12',
    priceYearly: '120',
  },
  {
    icon: '/src/assets/images/icon-pro.svg',
    title: 'Pro',
    priceMonthly: '15',
    priceYearly: '150',
  },
];
