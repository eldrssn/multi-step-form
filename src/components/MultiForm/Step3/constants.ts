export const addOns: Record<string, string>[] = [
  {
    title: 'Online service',
    name: 'service',
    description: 'Access to multiplayer games',
    priceYearly: '10',
    priceMonthly: '1',
  },
  {
    title: 'Larger storage',
    name: 'storage',
    description: 'Extra 1TB of cloud save',
    priceYearly: '20',
    priceMonthly: '2',
  },
  {
    title: 'Customizable Profile',
    name: 'profile',
    description: 'Custom theme on your profile',
    priceYearly: '20',
    priceMonthly: '2',
  },
];
