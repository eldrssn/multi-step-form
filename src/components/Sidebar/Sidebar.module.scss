@import '/src/styles/breakpoints.scss';

.container {
	background-image: url('/images/bg-sidebar-desktop.svg');
	background-repeat: no-repeat;
	background-size: cover;	
	min-width: 275px;
	height: 570px;

	padding-top: 40px;
	padding-left: 30px;
	padding-right: 30px;

	display: flex;
	flex-direction: column;
	gap: 30px;
}

@media screen and (max-width: $mobile) {
  .container {
    flex-direction: row;
    background-image: url('/images/bg-sidebar-mobile.svg');
    min-width: 100vw;
    width: auto;
    height: auto;
    gap: 15px;
    justify-content: center;
    padding: 30px 0;
    padding-bottom: 100px;
  }
}