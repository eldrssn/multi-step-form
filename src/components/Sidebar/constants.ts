export const steps = [
  {
    number: 1,
    title: 'Your info',
  },
  {
    number: 2,
    title: 'Select plan',
  },
  {
    number: 3,
    title: 'Add-ons',
  },
  {
    number: 4,
    title: 'Summary',
  },
];
