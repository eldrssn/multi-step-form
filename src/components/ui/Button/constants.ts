export const text = {
  back: 'Go Back',
  next: 'Next Step',
  confirm: 'Confirm',
};
