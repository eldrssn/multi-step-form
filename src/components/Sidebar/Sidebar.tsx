import { FC } from 'react';
import { Step } from '../ui/Step';
import { steps } from './constants';
import { TSidebar } from './types';
import styles from './Sidebar.module.scss';

const Sidebar: FC<TSidebar> = ({ step }) => (
  <aside className={styles.container}>
    {steps.map((item) => (
      <Step step={item} key={item.number} isActive={step === item.number} />
    ))}
  </aside>
);

export { Sidebar };
