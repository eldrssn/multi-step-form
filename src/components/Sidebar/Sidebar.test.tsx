import { render } from '@testing-library/react';
import { Sidebar } from './Sidebar';
import { steps } from './constants';

describe('Sidebar', () => {
  test('renders Sidebar component', () => {
    const { container } = render(<Sidebar />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('renders all steps in Sidebar', () => {
    const { getByText } = render(<Sidebar />);

    steps.forEach((step) => {
      expect(getByText(step.title)).toBeInTheDocument();
    });
  });
});
