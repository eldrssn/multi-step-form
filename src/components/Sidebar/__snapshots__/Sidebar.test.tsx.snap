// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders Sidebar component 1`] = `
<div>
  <aside
    class="container"
  >
    <div
      class="box box_active"
    >
      <div
        class="number"
      >
        1
      </div>
      <div
        class="info"
      >
        <p
          class="info_number"
        >
          Step 
          1
        </p>
        <p
          class="info_title"
        >
          Your info
        </p>
      </div>
    </div>
    <div
      class="box"
    >
      <div
        class="number"
      >
        2
      </div>
      <div
        class="info"
      >
        <p
          class="info_number"
        >
          Step 
          2
        </p>
        <p
          class="info_title"
        >
          Select plan
        </p>
      </div>
    </div>
    <div
      class="box"
    >
      <div
        class="number"
      >
        3
      </div>
      <div
        class="info"
      >
        <p
          class="info_number"
        >
          Step 
          3
        </p>
        <p
          class="info_title"
        >
          Add-ons
        </p>
      </div>
    </div>
    <div
      class="box"
    >
      <div
        class="number"
      >
        4
      </div>
      <div
        class="info"
      >
        <p
          class="info_number"
        >
          Step 
          4
        </p>
        <p
          class="info_title"
        >
          Summary
        </p>
      </div>
    </div>
  </aside>
</div>
`;
